src/index.tsx

```diff
@@ -13,30 +13,57 @@ const colorSchemes = ['light', 'dark']
 const MEDIA = '(prefers-color-scheme: dark)'
 const isServer = typeof window === 'undefined'
 const ThemeContext = createContext<UseThemeProps | undefined>(undefined)
-const defaultContext: UseThemeProps = { setTheme: _ => {}, themes: [] }
+const defaultContext: UseThemeProps = { setTheme: _ => { }, themes: [] }
 
 export const useTheme = () => useContext(ThemeContext) ?? defaultContext
 
-export const ThemeProvider: React.FC<ThemeProviderProps> = props => {
-  const context = useContext(ThemeContext)
-
-  // Ignore nested context providers, just passthrough children
-  if (context) return <Fragment>{props.children}</Fragment>
-  return <Theme {...props} />
-}
-
-const Theme: React.FC<ThemeProviderProps> = ({
+export const ThemeProvider: React.FC<ThemeProviderProps> = ({
   forcedTheme,
   disableTransitionOnChange = false,
   enableSystem = true,
   enableColorScheme = true,
   storageKey = 'theme',
-  themes = ['light', 'dark'],
+  themes = enableSystem ? ['light', 'dark', 'system'] : ['light', 'dark'],
   defaultTheme = enableSystem ? 'system' : 'light',
   attribute = 'data-theme',
   value,
   children,
   nonce
+}) => {
+  const context = useContext(ThemeContext)
+
+  // Ignore nested context providers, just passthrough children
+  if (context) return <Fragment>{children}</Fragment>
+  return (
+    <Theme
+      forcedTheme={forcedTheme}
+      disableTransitionOnChange={disableTransitionOnChange}
+      enableSystem={enableSystem}
+      enableColorScheme={enableColorScheme}
+      storageKey={storageKey}
+      themes={themes}
+      defaultTheme={defaultTheme}
+      attribute={attribute}
+      value={value}
+      nonce={nonce}
```

Comment:
Hmm no, I'd rather not. Spreading is fine here since it just composes another component with the exact same props.